h/api/nipsa/worker.py

```diff
@@ -0,0 +1,69 @@
+"""Worker functions for the NIPSA feature."""
+import json
+
+import annotator
+
+from h.api import search
+from h.api.nipsa import search as nipsa_search
+
+
+def add_nipsa_action(annotation):
+    """Return an Elasticsearch action for adding NIPSA to the annotation."""
+    return {
+        "_op_type": "update",
+        "_index": annotator.es.index,
+        "_type": "annotation",
+        "_id": annotation["_id"],
+        "doc": {"not_in_public_site_areas": True}
+    }
+
+
+def remove_nipsa_action(annotation):
+    """Return an Elasticsearch action to remove NIPSA from the annotation."""
+    return {
+        "_op_type": "update",
+        "_index": annotator.es.index,
+        "_type": "annotation",
+        "_id": annotation["_id"],
+        "script": "ctx._source.remove(\"not_in_public_site_areas\")"
+    }
+
+
+def add_or_remove_nipsa(userid, action, es_client):
```

Comment:
The fact that we use `add` and `remove` here makes me think we definitely should change `nipsa` and `unnipsa` in the logic module.